File: launcher/driver-api/jvm/src/main/java/dev/argon/driver/api/CompilerDriver.java
package dev.argon.driver.api;

import dev.argon.backend.api.metadata.BackendMetadata;
import dev.argon.driver.api.command.DriverCommand;

import java.util.List;

public interface CompilerDriver {
	BackendMetadata parseMetadata(String metadata) throws BackendMetadataParseException;
	DriverCommand parseCommandLineArguments(List<BackendMetadata> backends, String[] args);
    int runCommand(CompilerDriverOptions options) throws Throwable;
}

File: launcher/driver/node/src/main/scala/dev/argon/driver/sjs/CompilerDriver.scala
package dev.argon.driver.sjs

import dev.argon.backend.sjs.metadata.BackendMetadata
import dev.argon.driver.sjs.command.DriverCommand

import scala.scalajs.js

trait CompilerDriver extends js.Object {
  def parseMetadata(metadata: String): BackendMetadata
  def parseCommandLineArguments(backends: js.Array[BackendMetadata], args: js.Array[String]): DriverCommand[String, String, String, String]
  def runCommand(options: CompilerDriverOptions): js.Promise[Int]
}
